src/components/MovieForm/MovieForm.tsx

```diff
@@ -0,0 +1,95 @@
+import './MovieForm.css'
+import React from 'react'
+
+type MovieFormProps = {
+    initialInfo?: {
+        title: string
+        releaseDate: string
+        movieUrl: string
+        rating: number
+        genres: string[]
+        runtime: number
+        overview: string
+    }
+    onSubmit: () => void
+}
+
+export function MovieForm({ initialInfo, onSubmit }: MovieFormProps) {
+    function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+        event.preventDefault()
+        onSubmit()
+    }
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <div className={'movie-form'}>
+                <div>
+                    <label htmlFor={'title'}>TITLE</label>
+                    <input
+                        id={'title'}
+                        type={'text'}
+                        defaultValue={initialInfo?.title}
+                        placeholder={'Movie Title'}
+                    ></input>
+                </div>
+                <div>
+                    <label htmlFor={'release-date'}>RELEASE DATE</label>
+                    <input
+                        id={'release-date'}
+                        type={'date'}
+                        defaultValue={initialInfo?.releaseDate}
+                    ></input>
+                </div>
+                <div>
+                    <label htmlFor={'movie-url'}>MOVIE URL</label>
+                    <input
+                        id={'movie-url'}
+                        type={'text'}
+                        defaultValue={initialInfo?.movieUrl}
+                        placeholder={'https://'}
+                    ></input>
+                </div>
+                <div>
+                    <label htmlFor={'rating'}>RATING</label>
+                    <input
+                        id={'rating'}
+                        type={'number'}
+                        defaultValue={initialInfo?.rating}
+                        placeholder={'0-10'}
+                    ></input>
+                </div>
+                <div>
+                    <label htmlFor={'genre'}>GENRE</label>
+                    <select id={'genre'}>
+                        {initialInfo?.genres?.map(genre => (
+                            <option key={genre} value={genre}>
+                                {genre}
+                            </option>
```

Comment:
I guess there must be options in this select even when no `initialInfo` object is provided. For example, when we are adding a new movie, we want to choose from the list of all possible genres. Also, since one movie can have multiple genres, a multiselect control should be used.

Reply:
I've tried the multiple option on select tag, but it renders an ugly list of genres without a dropdown. I'll find some libs that provide better components and use them